src/org/zaproxy/zap/extension/exportreport/Export/ExportReport.java

```diff
@@ -0,0 +1,247 @@
+				f_view = ReportExport.TRANSFORMATION(view, logger, absolutePath, (path + fileName + ".xml"), XML_PATH, mergeXSL);
+				show = true;
+				break;
+			case Utils.json:
+				view.showMessageDialog(Constant.messages.getString("exportreport.message.notice.json"));
+				f_view = ReportExport.TRANSFORMATION(view, logger, absolutePath, (path + fileName + ".xml"), XML_PATH, mergeXSL);
+				f_view = null;
+				f_view = ReportExport.JSON_EXPORT(view, logger, absolutePath, absolutePath,(path + fileName + ".xml"));
+				deleteFile(path + fileName + ".xml", "The merged XML file: ");
+				break;
+			case Utils.pdf:
+				view.showMessageDialog(Constant.messages.getString("exportreport.message.notice.pdf"));
+				f_view = null;
+				break;
+			case Utils.doc:
+				view.showMessageDialog(Constant.messages.getString("exportreport.message.notice.doc"));
+				f_view = null;
+				break;
+			default:
+				break;
+			}
+			deleteFile(XML_PATH, "The data dump file: ");
+			try {
+				if (!(f_view == null) && (show == true))
+				{
+					DesktopUtils.openUrlInBrowser(f_view.toURI());					
+				}
+			} catch (Exception e) {
+				logger.error(e.getMessage(), e);
+				view.showWarningDialog(Constant.messages.getString("exportreport.message.error.file.open"));
+			}
+			return null;
+		}
+
+		/**
+		 * Executed in event dispatching thread
+		 */
+		@Override
+		public void done() {
+			Toolkit.getDefaultToolkit().beep();
+		}
+	}
+
+	public void generateReport(ViewDelegate view, ExtensionExportReport extension) {
+		FileList list = generateFileList();
+		try {
+			if (fc == null) {
+				fc = generateWriteableFileChooser(fc, list);
+				fc.addPropertyChangeListener(WritableFileChooser.FILE_FILTER_CHANGED_PROPERTY,
+						new PropertyChangeListener() {
+							@Override
+							public void propertyChange(PropertyChangeEvent evt) {
+								ReportFilter filter = (ReportFilter) evt.getNewValue();
+								String extension = (filter.getExtensionByDescription(filter.getDescription())
+										.length() == 0 ? ""
+												: "." + filter.getExtensionByDescription(filter.getDescription()));
+								fc.setSelectedFile(new File(extension));
+							}
+						});
+			}
+
+			int rc = fc.showSaveDialog(View.getSingleton().getMainFrame());
+
+			boolean bool = false;
+			String fileExtension = "";
+			while (rc == WritableFileChooser.APPROVE_OPTION && !bool) {
+				fileExtension = list.compareExtension(fc.getSelectedFile().getName());
+				if (fileExtension.length() > 0) {
+					bool = true;
+				}
+				if (!bool) {
+					//Determine in v2.0 which method to use for display of error messages based on feedback.
+					//JOptionPane.showMessageDialog(null,
+					//		"The file " + fc.getSelectedFile() + " is not a valid destination file.", "Open Error",
+					//		JOptionPane.ERROR_MESSAGE);
+					view.showWarningDialog(Constant.messages.getString("exportreport.message.error.file.destination", fc.getSelectedFile()));
+					rc = fc.showSaveDialog(View.getSingleton().getMainFrame());
```

Comment:
Delayed this one temporarily since i am working off of an older source of `org.zaproxy.zap.view.widgets.WritableFileChooser`, i'll make sure this change gets into the next iteration though if that's okay since it doesn't negatively affect it currently. Is that okay?